src/platform_linux/laio.c

```diff
@@ -378,34 +518,58 @@ laio_cleanup(io_handle *ioh, uint64 count)
    uint64          i;
    int             status;
 
+   threadid tid = platform_get_tid();
+
    io = (laio_handle *)ioh;
 
    // Check for completion of up to 'count' events, one event at a time.
    // Or, check for all outstanding events (count == 0)
    for (i = 0; ((count == 0) || (i < count)); i++) {
-      status = io_getevents(io->ctx, 0, 1, &event, NULL);
+      status = io_getevents(io->ctx[tid], 0, 1, &event, NULL);
       if (status < 0) {
-         platform_error_log("io_getevents[%lu], count=%lu, "
-                            "failed with errorno=%d: %s\n",
-                            i,
-                            count,
-                            -status,
-                            strerror(-status));
+         platform_error_log(
+            "%s(): OS-pid=%d, tid=%lu, io_getevents[%lu], count=%lu, "
+            "failed with errorno=%d: %s\n",
+            __func__,
+            getpid(),
+            tid,
+            i,
+            count,
+            -status,
+            strerror(-status));
          i--;
-         continue;
+
+         // We got a hard-error probably because some code-flow messed-up
+         // using the per-thread IO-context. No point in trying again and
+         // again if we are checking for completion of all outstanding
+         // events.
+         if (count == 0) {
+            break;
+         } else {
+            // Retry a few times for a finite # of pending events
+            continue;
```

Comment:
NOTE to myself: I think this re-try logic can come out now. I had added it during code-dev stabilization. This area is quite firm now.

Comments?

Reply:
@rtjohnso - Remove this new changed behaviour and revert to the old code flow.

Reply:
Done.